File: src/main/java/com/example/moexbondservice/service/utils/MoexBondValidator.java
package com.example.moexbondservice.service.utils;

import com.example.moexbondservice.dto.BondDto;
import com.example.moexbondservice.exception.LimitMoexRequestExceededException;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@Slf4j
public class MoexBondValidator {

    public void validateMoexLimitation(List<BondDto> bonds) {
        if (bonds.isEmpty()) {
            log.error("Exceeded limits of MOEX requests");
            throw new LimitMoexRequestExceededException("Exceeded limits of MOEX requests");
        }
    }

}

File: src/main/java/com/example/moexbondservice/service/utils/MoexBondValidator.kt
package com.example.moexbondservice.service.utils

import com.example.moexbondservice.dto.BondDto
import com.example.moexbondservice.exception.LimitMoexRequestExceededException
import mu.KotlinLogging
import org.springframework.stereotype.Service

@Service
class MoexBondValidator {
    private val log = KotlinLogging.logger{}
    fun validateMoexLimitation(bonds: List<BondDto?>?) {
        if (bonds!!.isEmpty()) {
            log.error("Exceeded limits of MOEX requests")
            throw LimitMoexRequestExceededException("Exceeded limits of MOEX requests")
        }
    }
}
